tests/utils.py

```diff
@@ -21,6 +22,102 @@
 from m2cgen import ast
 
 
+class StatsmodelsSklearnLikeWrapper(BaseEstimator, RegressorMixin):
+    def __init__(self, model, params):
+        self.model = model
+        self.params = params
+        self.__class__.__name__ = model.__name__
+
+    def fit(self, X, y):
+        init_params = self.params.get("init", {})
+        self.fit_intercept_ = init_params.pop("fit_intercept", False)
+        if self.fit_intercept_:
+            X = sm.add_constant(X)
+        est = self.model(y, X, **init_params)
+        if "fit_regularized" in self.params:
+            self.fitted_model_ = est.fit_regularized(
+                **self.params["fit_regularized"])
+        else:
+            self.fitted_model_ = est.fit(**self.params.get("fit", {}))
+        self.__class__.__name__ = type(self.fitted_model_).__name__
```

Comment:
This is something I don't fully follow. I assume that the type of the fitted model is different from the one passed into the constructor. I think I get it when we do this in constructor - so that `get_assembler_cls` can pick the right assembler, but why do we do this here? 

**UPD** I guess this is needed to make a model name in E2E tests look nicer (eg. OLS, GLS, etc.). Please correct me if I'm wrong.

Reply:
Perhaps worth leaving an explanatory comment in both places, so that we don't have same questions 6 months from now :D

Reply:
> UPD I guess this is needed to make a model name in E2E tests look nicer (eg. OLS, GLS, etc.). Please correct me if I'm wrong.

You are absolutely right!

>Perhaps worth leaving an explanatory comment in both places, so that we don't have same questions 6 months from now :D

OK!